src/ui/controllers/skills/index.ts

```diff
@@ -1,143 +1,243 @@
 import * as express from 'express'
-import {NextFunction} from "express"
 import * as model from 'lib/model'
+import * as registry from "lib/registry"
 import * as skillsApi from "lib/service/skills"
-import {Choice, Question} from "lib/service/skills/api"
+
+import {
+	Answer,
+	AnswerSubmission, AreaOfWorkKeysInterface,
+	Question,
+	Quiz, QuizHistory,
+	QuizMetadata, QuizType,
+	SelectedAnswer,
+	SelectedAnswers
+} from "lib/service/skills/api"
 import * as template from 'lib/ui/template'
+import {formatAnswerSubmissionDate, saveSession} from "./helpers"
 
-export function introduction(req: express.Request, res: express.Response) {
-	let show: boolean = false
+export async function introduction(req: express.Request, res: express.Response) {
 	const user = req.user as model.User
-	// @ts-ignore
-	if (user.areasOfWork && user.areasOfWork.length > 0) {
-		show = user.areasOfWork[1].toLocaleLowerCase() === 'policy'
+	if (!(user.areasOfWork && user.areasOfWork.length > 0) || !user.organisationalUnit!.id) {
+		showQuizScreen(req, res, false)
+		return
 	}
-	res.send(template.render('skills/introduction', req, res, {show}))
+
+	const professionId = parseInt(user.areasOfWork[0], 10)
```

Comment:
do you have access to civil servant object, if not you can fire a get and grab one. That will have profession and organisation information. a get to this csrs endpoint `/civilServants/me`  will get the needed. then you can carry it through the journey.

Reply:
The user object is exactly that, it calls "/civilServants/me" and saves it to the user object, and it contains all the information (profession/organisation), it's commonly used in different places in the app too.

Reply:
Yes. I agree. It's a pity we need to access the professionId from an array. It would have been nicer if we were able to, say, access it like user.profession.primary. We can resolve this discussion.